src/ape/managers/chain.py

```diff
@@ -190,15 +193,15 @@ def range(
         # Note: the range `stop_block` is a non-inclusive stop, while the
         #       `.query` method uses an inclusive stop, so we must adjust downwards.
         results = self.query("*", start_block=start, stop_block=stop - 1, step=step)  # type: ignore
```

Comment:
Pull in all of the logic to call `self.query_manager.query(query)` and yield from that instead

Then update `self.query` to call this function and convert to a dataframe instead